framework/db/BaseActiveRecord.php

```diff
@@ -522,14 +542,14 @@ public function markAttributeDirty($name)
      */
     public function isAttributeChanged($name, $identical = true)
     {
-        if (isset($this->_attributes[$name], $this->_oldAttributes[$name])) {
+        if (isset($this->_oldAttributes[$name]) && (($value = $this->getAttribute($name)) !== null)) {
             if ($identical) {
-                return $this->_attributes[$name] !== $this->_oldAttributes[$name];
+                return $value !== $this->_oldAttributes[$name];
             } else {
-                return $this->_attributes[$name] != $this->_oldAttributes[$name];
+                return $value != $this->_oldAttributes[$name];
             }
         } else {
-            return isset($this->_attributes[$name]) || isset($this->_oldAttributes[$name]);
+            return ($this->getAttribute($name) != null) || isset($this->_oldAttributes[$name]);
```

Comment:
Yes.

(That logic joke, always fun.)

It definitely should be `!==`. Also if this passes test(s) then test coverage is not ideal.

Will look into this over the weekend probably -- the tests that is, will fix the comparison now.